File: backend/src/models/boletim.py
from pydantic import BaseModel
from typing import Optional, List, Dict, Any
from datetime import datetime


class BoletimCreate(BaseModel):
    acidente_id: str
    numero_processo: Optional[str] = None
    modo_criacao: str = "GERADO_SISTEMA"
    observacoes: Optional[str] = None
    vitimas_info: Optional[List[Dict[str, Any]]] = []
    veiculos_info: Optional[List[Dict[str, Any]]] = []
    testemunhas: Optional[List[Dict[str, Any]]] = []


class BoletimResponse(BaseModel):
    boletim_id: str
    acidente_id: str
    numero_processo: str
    modo_criacao: str
    observacoes: Optional[str] = None
    vitimas_info: List[Dict[str, Any]]
    veiculos_info: List[Dict[str, Any]]
    testemunhas: List[Dict[str, Any]]
    arquivo_url: Optional[str] = None
    created_at: datetime
    created_by: Optional[str] = None

File: backend/src/models/Boletim.js
const mongoose = require('mongoose');

const boletimSchema = new mongoose.Schema({
  boletim_id: { type: String },
  acidente_id: { type: String },
  numero_processo: { type: String },
  modo_criacao: { type: String, default: 'GERADO_SISTEMA' },
  observacoes: { type: String, default: '' },
  vitimas_info: [{ type: mongoose.Schema.Types.Mixed }],
  veiculos_info: [{ type: mongoose.Schema.Types.Mixed }],
  testemunhas: [{ type: mongoose.Schema.Types.Mixed }],
  arquivo_url: { type: String, default: null },
  created_at: { type: Date, default: Date.now },
  created_by: { type: String, default: '' }
}, { collection: 'boletins', timestamps: false });

module.exports = mongoose.model('Boletim', boletimSchema);
